uninstall.php

```diff
@@ -0,0 +1,173 @@
+<?php
+/**
+ * Distributor uninstall script.
+ *
+ * @package distributor
+ * @since   x.x.x
+ */
+
+// phpcs:disable
+
+if ( ! defined( 'WP_UNINSTALL_PLUGIN' ) ) {
+	die;
+}
+
+/*
+ * Only remove ALL data if DT_REMOVE_ALL_DATA constant is set to true in user's
+ * wp-config.php. This is to prevent data loss when deleting the plugin from the backend
+ * and to ensure only the site owner can perform this action.
+ */
+if ( defined( 'DT_REMOVE_ALL_DATA' ) && true === DT_REMOVE_ALL_DATA ) {
+	global $wpdb;
+
+	/**
+	 * Function to delete all relevant data from the site.
+	 */
+	function dt_delete_data() {
+		global $wpdb;
+
+		// Delete post meta and posts of type 'dt_subscription'.
+		$subscription_post_ids = $wpdb->get_col(
+			$wpdb->prepare(
+				"SELECT ID FROM $wpdb->posts WHERE post_type = %s",
+				'dt_subscription'
+			)
+		);
+
+		if ( ! empty( $subscription_post_ids ) ) {
+			$ids_string = implode( ',', array_map( 'intval', $subscription_post_ids ) );
+
+			// Delete subscription meta.
+			$wpdb->query(
+				$wpdb->prepare(
+					"DELETE FROM $wpdb->postmeta WHERE post_id IN ($ids_string)"
+				)
+			);
+
+			// Delete subscription posts.
+			$wpdb->query(
+				$wpdb->prepare(
+					"DELETE FROM $wpdb->posts WHERE ID IN ($ids_string)"
+				)
+			);
+
+			// Clear the post cache.
+			wp_cache_set_posts_last_changed();
+			wp_cache_delete_multiple( $subscription_post_ids, 'posts' );
+			wp_cache_delete_multiple( $subscription_post_ids, 'post_meta' );
+		}
+
+		// Delete relevant options from the options table.
+		delete_site_options();
+	}
+
+	/**
+	 * Delete all relevant options from the options table.
+	 */
+	function delete_site_options() {
+		global $wpdb;
+
+		$option_prefixes = array(
+			'dt_',
+			'_transient_dt_',
+			'_transient_timeout_dt_',
+		);
+
```

Comment:
This will need to include the site_transient prefixes for `! multisite()` -- they're stored in the options table but with still use the site prefix.

Reply:
I was under the impression that `_site_transient_dt_` and `_site_transient_timeout_dt_` are used exclusively in multisite networks. That’s why I’m not quite sure why we should include them in this context, and that also with `! multisite()` condition!? Shouldn't we use a `multisite()` condition? Side note: we are already removing them from the `sitemeta` table.

Reply:
Site transients can be used on single sites too but are stored in the options table rather than the site meta table with the `_site_transient` prefix. They use the `*_site_option` functions which eventually flow through to the `*_option` functions on single site installs.